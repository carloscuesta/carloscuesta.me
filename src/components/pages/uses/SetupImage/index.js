// @flow
import React, { type Element } from 'react'

import styles from './styles.module.css'

const SetupImage = (): Element<'img'> => (
  <img
    className={`lazyload ${styles.image}`}
    data-src='https://res.cloudinary.com/carloscuesta/image/upload/v1628605093/setup_qdzbaw.jpg'
    src='https://res.cloudinary.com/carloscuesta/image/upload/s--blnqMrZY--/c_scale,e_blur:100,q_10,w_200/v1628605093/setup_qdzbaw.jpg'
    width={600}
  />
)

export default SetupImage
