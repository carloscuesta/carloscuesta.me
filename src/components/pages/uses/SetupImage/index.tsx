
import styles from './styles.module.css'

const SetupImage = () => (
  <img
    alt='Carlos Cuesta workspace setup'
    className={`lazyload ${styles.image}`}
    data-src='https://res.cloudinary.com/carloscuesta/image/upload/v1659775881/carloscuesta-setup_z1azyb.jpg'
    src='https://res.cloudinary.com/carloscuesta/image/upload/s--pm3d7vcD--/c_scale,e_blur:200,q_20,w_500/v1659775881/carloscuesta-setup_z1azyb.jpg'
    width={2016}
  />
)

export default SetupImage
