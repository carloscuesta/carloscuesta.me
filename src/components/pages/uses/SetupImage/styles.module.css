.image {
  filter: blur(8px);
  /* https://stackoverflow.com/questions/27532270/css-blur-weird-shadow-effect */
  transform: translateZ(0);
  transition: filter 200ms ease-in-out;
  border-radius: 4px;
  max-width: 100%;
}

.image:global(.lazyloaded) {
  filter: blur(0);
}
