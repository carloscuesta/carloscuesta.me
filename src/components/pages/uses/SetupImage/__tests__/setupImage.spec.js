import renderer from 'react-test-renderer'

import SetupImage from '../index'

describe('SetupImage', () => {
  it('should match SetupImage component', () => {
    const wrapper = renderer.create(<SetupImage />)

    expect(wrapper).toMatchSnapshot()
  })
})
