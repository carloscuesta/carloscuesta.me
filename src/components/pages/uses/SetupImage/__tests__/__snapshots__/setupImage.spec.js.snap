// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetupImage should match SetupImage component 1`] = `
<img
  className="lazyload image"
  data-src="https://res.cloudinary.com/carloscuesta/image/upload/v1628605093/setup_qdzbaw.jpg"
  src="https://res.cloudinary.com/carloscuesta/image/upload/s--blnqMrZY--/c_scale,e_blur:100,q_10,w_200/v1628605093/setup_qdzbaw.jpg"
  width={600}
/>
`;
